orttraining/orttraining/test/graph/optimizer_graph_builder_test.cc

```diff
@@ -96,21 +101,50 @@ Status SetUpBaseGraph(Graph& graph) {
   return graph.Resolve(resolve_options);
 }
 
-std::unordered_map<std::string, OptimizerNodeConfig> GetOptInfoMap() {
+std::unordered_map<std::string, OptimizerNodeConfig> GetOptInfoMap(const std::string& optim_name) {
   std::unordered_map<std::string, OptimizerNodeConfig> result{};
   std::transform(
       k_weight_names.begin(), k_weight_names.end(), std::inserter(result, result.end()),
-      [](const std::string& weight_name) {
+      [&optim_name](const std::string& weight_name) {
         return std::make_pair(
-            weight_name, OptimizerNodeConfig{k_optimizer_op_name, nullptr, "Learning_Rate", {}});
+            weight_name, OptimizerNodeConfig{optim_name, nullptr, "Learning_Rate", {}});
       });
+
   return result;
 }
 
+std::unordered_map<std::string, OptimizerNodeConfig> GetOptInfoMap() {
+  return GetOptInfoMap(k_adam_optimizer_op_name);
+}
+
 OptimizerBuilderRegistry& GetOptimizerBuilderRegistry() {
   return OptimizerBuilderRegistry::GetInstance();
 }
 
+void VerifyTensorValue(const ONNX_NAMESPACE::TensorProto* tensor, float expected_val) {
+  float tensor_value;
+  ASSERT_TRUE(tensor->dims_size() == 1);
+  ASSERT_TRUE(tensor->dims(0) == 1);
+  if (tensor->has_raw_data()) {
+    memcpy(&tensor_value, tensor->raw_data().data(), sizeof(float));
+  } else {
+    tensor_value = *(tensor->float_data().data());
+  }
+  ASSERT_EQ(tensor_value, expected_val);
+}
+
+void VerifyTensorValue(const ONNX_NAMESPACE::TensorProto* tensor, int64_t expected_val) {
```

Comment:
can this function be templatized?

Reply:
it could be but it will have to be specialized for both types anyway, so there might be little to gain from templatizing